kube/services/amanuensis/amanuensis-deploy.yaml

```diff
@@ -0,0 +1,204 @@
+              memory: 2400Mi
+        command: ["/bin/bash"]
+        args:
+          - "-c"
+          - |
+            echo "${AMANUENSIS_PUBLIC_CONFIG:-""}" > "/var/www/amanuensis/amanuensis-config-public.yaml"
+            python /var/www/amanuensis/yaml_merge.py /var/www/amanuensis/amanuensis-config-public.yaml /var/www/amanuensis/amanuensis-config-secret.yaml > /var/www/amanuensis/amanuensis-config.yaml
+            #bash /amanuensis/dockerrun.bash && if [[ -f /dockerrun.sh ]]; then /dockerrun.sh; fi
+            bash /dockerrun.sh
+      - name: nginx-prometheus-exporter-wrapper
+        GEN3_NGINX_PROMETHEUS_EXPORTER_WRAPPER_IMAGE|-image: quay.io/cdis/nginx-prometheus-exporter-wrapper:pybase3-1.4.0-|
+        command: ["/bin/bash"]
+        args:
+          - "-c"
+          - |
+            attempt=0
+            maxAttempts=12
+            while true
+            do
+              nginx_status_response=$(curl -L -s -o /dev/null -w "%{http_code}" -X GET http://localhost/nginx_status)
+              echo "initializing nginx-prometheus-exporter..."
+              echo "nginx_status_response: $nginx_status_response"
+              if [ "$nginx_status_response" == 200 ]; then
+                echo "nginx_status is finally up and running. Proceed with the metrics exporting..."
+                /usr/bin/exporter -nginx.scrape-uri http://127.0.0.1/nginx_status
+              else
+                echo "The nginx_status endpoint is not ready yet... attempt #${attempt}"
+                sleep 5
+                if [ $attempt -eq $maxAttempts ];then
+                  echo "The nginx_status endpoint was never initialized properly, keep the sidecar running in sleep mode to prevent the entire pod from failing..."
+                  sleep infinity
+                fi
+              fi
+              attempt=$(( $attempt + 1 ));
+            done
+        ports:
+        - containerPort: 9113
+      - name: uwsgi-exporter
+        GEN3_UWSGI_EXPORTER_IMAGE|-image: registry.hub.docker.com/timonwong/uwsgi-exporter:v1.0.0-|
+        args: ["--stats.uri", "http://127.0.0.1/uwsgi_status"]
+        ports:
+        - containerPort: 9117
+      # initContainers:
+      # - name: amanuensis-init
+        # GEN3_AMANUENSIS_IMAGE
+        # imagePullPolicy: Always
+        # env:
+        #   - name: PYTHONPATH
+        #     value: /var/www/amanuensis
+        #   - name: AMANUENSIS_PUBLIC_CONFIG
+        #     valueFrom:
+        #       configMapKeyRef:
+        #         name: manifest-amanuensis
+        #         key: amanuensis-config-public.yaml
+        #         optional: true
+        # volumeMounts:
+        #   - name: "config-volume"
+        #     readOnly: true
+        #     mountPath: "/var/www/amanuensis/amanuensis-config-secret.yaml"
+        #     subPath: amanuensis-config.yaml
+        #   - name: "yaml-merge"
+        #     readOnly: true
+        #     mountPath: "/var/www/amanuensis/yaml_merge.py"
+        #     subPath: yaml_merge.py
```

Comment:
Do you know why in fence there is this initContainer to run DB migration but there is also a job to create the DB? Just trying to understand if there is a reason for it or I can just keep either the job or this initContainer deployment

Reply:
Yea the container is supposed to be there to automatically handle migrations, if the flag is set in fence-config and a new db scheme is coming. The db create job will do the initial setup, but the init container will perform migrations to new schemas in already running db's. I don't know if you plan to do many migrations, but if you do you might consider including an init container like this. We also actually have a fence-db-migrate job, which I just noticed you didn't recreate here, which can perform the migrations if you don't want to worry about automatic migrations.